backend/src/main/kotlin/fr/gouv/dgampa/rapportnav/infrastructure/database/model/mission/action/ActionAntiPollutionModel.kt

```diff
@@ -74,9 +82,22 @@ data class ActionAntiPollutionModel(
             longitude = action.longitude,
             detectedPollution = action.detectedPollution,
             pollutionObservedByAuthorizedAgent = action.pollutionObservedByAuthorizedAgent,
-            diversionCarriedOut = action.diversionCarriedOut
+            diversionCarriedOut = action.diversionCarriedOut,
+            isAntiPolDeviceDeployed = action.isAntiPolDeviceDeployed,
+            isSimpleBrewingOperationDone = action.isSimpleBrewingOperationDone
         )
     }
+
+    override fun hashCode(): Int {
+        return Objects.hash(id)
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+        other as ActionAntiPollutionModel
+        return id == other.id
+    }
```

Comment:
Pourquoi t'as rajouté ça ?
J'ai jamais trop compris ce délire avec les classes et le hashcode mais ça m'a dejap ris la tête dans le passé

Reply:
en écrivant le test sur le class  `ActionAntiPollutionModel` j'avais cette erreur: 
`Caused by: java.lang.NullPointerException: Cannot invoke "java.util.UUID.hashCode()" because "this.id" is null`
Spring arrivait pas à instancier la classe a cause du null Pointer de la méthode hashCode.
Donc j'ai override la méthode hashCode manuellement.
Après quand on override la méthode hashCode on est obligé de redéfinir la méthode equals, parce que equals se base du hashCode.

D'où la fameuse phrase: 
2 classes qui sont égales ont le même hashCode, mais 2 classes avec le même hashCode ne sont pas forcément égales.